api/client/event/event_stream.go

```diff
@@ -93,6 +93,8 @@ func (h *EventStream) Subscribe(eventsChannel chan<- *Event) {
 			EventType: EventConnectionError,
 			Data:      []byte(errors.Wrap(err, client.ErrConnectionIssue.Error()).Error()),
 		}
+		close(eventsChannel)
```

Comment:
Thanks for this PR and report, I'm a bit concerned about the `close(eventsChannel)` here, as we use it when we restart the stream here https://github.com/prysmaticlabs/prysm/blob/77f10b9e0ee49634cc00548f0148adaa1a7bfbd5/validator/client/runner.go#L327

perhaps we should just be returning here and not closing the events stream. we'd like to do some testing before to make sure we have the right solution.